pkg/sysconfwatch/sysconfwatch_linux.go

```diff
@@ -103,16 +103,21 @@ func (c *SysConfWatchController) Run(ctx context.Context, ready chan<- struct{},
 		select {
 		case <-ticker.C:
 			// Check the IP change
-			currentIP, _ := util.GetHostIP()
+			currentIP, err := util.GetHostIP()
+			if err != nil {
+				klog.Warningf("cannot find an host IP: %v", err)
+				os.Exit(1)
+				return nil
+			}
 			if c.NodeIP != currentIP {
 				klog.Warningf("IP address has changed from %q to %q, restarting MicroShift", c.NodeIP, currentIP)
-				os.Exit(0)
+				os.Exit(1)
```

Comment:
If we automatically detect the default node IP differently in a way that allows that IP to change when the host's network settings change, then when the host goes on or off of a network MicroShift will exit when the default route IP changes and then fail to start, exiting here, because the previous node IP does not match the "current" IP value.

We have a way for the user to specify the node IP using the configuration file already. Can they use that setting with an IP that does not have the default route? Will kubelet accept that IP as the node IP?

If so, do we need to change the way we automatically detect the node IP? Maybe I'm misunderstanding the intent of the change?

Reply:
This patch doesn't fix the conflict between the configured NodeIP and the detected one. It's only for fixing [USHIFT-1551](https://issues.redhat.com//browse/USHIFT-1551).

> We have a way for the user to specify the node IP using the configuration file already. Can they use that setting with an IP that does not have the default route? Will kubelet accept that IP as the node IP?

Kubelet can use any non-loopback host IP as the NodeIP. The problem is on the `sysconfigwatch` side. If the IP detected by `sysconfigwatch` differs from the NodeIP in the Microshift config file, `sysconfigwatch` will always restart microshift.

I can update the PR to also fix the `NodeIP` flag issue.

Reply:
> I can update the PR to also fix the `NodeIP` flag issue.

Let me re-read this today, now that I see your answer. I may have misunderstood what you were trying to do.

Reply:
I've added the missing part. Now if users config `NodeIP` flag, `sysconfwatch` will only check the existence of the IP on the host. If yes, the provided NodeIP will always be used. Otherwise, Microshift will exit.